FiniteStateMachine/src/statemachine/State.java

```diff
@@ -2,11 +2,9 @@
 
 import org.jetbrains.annotations.NotNull;
 
-import java.util.EnumSet;
+import java.util.Set;
 
 
-public interface State<E extends Enum<E> & State<E>> {
```

Comment:
But nothing about this class requires the states to be any of those things. Again, it's an implementation detail. If your specific implementation of `StateMachine` needs an enum for its own reasons, it can require that with its type parameter binding.